adapters/InMobi/InMobiAdapter/GADMAdapterInMobiBannerAd.m

```diff
@@ -0,0 +1,194 @@
+    }];
+}
+
+- (void)requestBannerWithSize:(GADAdSize)adSize {
+    long long placementId =
+    [_bannerAdConfig.credentials.settings[GADMAdapterInMobiPlacementID] longLongValue];
+    
+    if (placementId == 0) {
+        NSError *error = GADMAdapterInMobiErrorWithCodeAndDescription(
+                                                                      GADMAdapterInMobiErrorInvalidServerParameters,
+                                                                      @"[InMobi] Error - Placement ID not specified.");
+        _bannerRenderCompletionHandler(nil, error);
+        return;
+    }
+    
+    if (_bannerAdConfig.isTestRequest) {
+        NSLog(@"[InMobi] Please enter your device ID in the InMobi console to recieve test ads from "
+              @"Inmobi");
+    }
+    
+    CGSize size = GADMAdapterInMobiSupportedAdSizeFromGADAdSize(adSize);
+    if (CGSizeEqualToSize(size, CGSizeZero)) {
+        NSString *description =
+        [NSString stringWithFormat:@"Invalid size for InMobi mediation adapter. Size: %@",
+         NSStringFromGADAdSize(adSize)];
+        NSError *error = GADMAdapterInMobiErrorWithCodeAndDescription(
+                                                                      GADMAdapterInMobiErrorBannerSizeMismatch, description);
+        _bannerRenderCompletionHandler(nil, error);
+        return;
+    }
+    
+    _adView = [[IMBanner alloc] initWithFrame:CGRectMake(0, 0, size.width, size.height)
+                                  placementId:placementId];
+    
+    // Let Mediation do the refresh.
+    [_adView shouldAutoRefresh:NO];
+    _adView.transitionAnimation = UIViewAnimationTransitionNone;
+    
+    GADInMobiExtras *extras = _bannerAdConfig.extras;
+    if (extras && extras.keywords) {
+        [_adView setKeywords:extras.keywords];
+    }
+    
+    GADMAdapterInMobiSetTargetingFromAdConfiguration(_bannerAdConfig);
+    NSDictionary<NSString *, id> *requestParameters =
+    GADMAdapterInMobiCreateRequestParametersFromAdConfiguration(_bannerAdConfig);
+    [_adView setExtras:requestParameters];
+    
+    _adView.delegate = self;
+    [_adView load];
+}
+
+- (void)stopBeingDelegate {
+    _adView.delegate = nil;
+}
+
+#pragma mark IMBannerDelegate methods
+
+- (void)bannerDidFinishLoading:(nonnull IMBanner *)banner {
+    NSLog(@"<<<<<ad request completed>>>>>");
+    _bannerAdEventDelegate = _bannerRenderCompletionHandler(self,nil);
+    [_bannerAdEventDelegate willPresentFullScreenView];
+}
+
+- (void)banner:(nonnull IMBanner *)banner didFailToLoadWithError:(nonnull IMRequestStatus *)error {
+    _bannerRenderCompletionHandler(nil, error);
+}
+
+- (void)banner:(nonnull IMBanner *)banner didInteractWithParams:(nonnull NSDictionary *)params {
+    NSLog(@"<<<< bannerDidInteract >>>>");
+    [_bannerAdEventDelegate reportClick];
+}
+
+- (void)userWillLeaveApplicationFromBanner:(nonnull IMBanner *)banner {
+    NSLog(@"<<<< bannerWillLeaveApplication >>>>");
```

Comment:
`"InMobi SDK will cause the user to leave the application from banner ad."`

Reply:
done